### PR TITLE
Feat/on this page

### PR DESCRIPTION
**Pull request by marianayovcheva**
Moving the _On this page_ component from public-website into the design-system. Details outlined here: https://github.com/citizensadvice/design-system/issues/1671

### CODE REVIEW

File: engine/app/components/citizens_advice_components/on_this_page.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+    def columns?
+      links.size > 3
+    end
+
+    def show_nested_links?
+      @show_nested_links
+    end
+
+    def children_exist?
```
Comment (marianayovcheva) [inline engine/app/components/citizens_advice_components/on_this_page.rb:40]: I'm not sure I've done the children_exist logic properly so it's worth checking

File: styleguide/components/on-this-page/on-this-page.stories.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+export const Example = () => templateExample;
+Example.parameters = { docs: { source: { code: templateExample } } };
+
+export const WithNestedLinks = () => templateWithNestedLinks;
+WithNestedLinks.parameters = {
+  docs: { source: { code: templateWithNestedLinks } },
+};
+WithNestedLinks.decorators = [decorateWithBehaviour];
```
Comment (marianayovcheva) [inline styleguide/components/on-this-page/on-this-page.stories.js:24]: In this file I had to add the decorator only on `WithNestedLinks`. It was running the js on the `Example` as well which caused problems - the nested links wouldn't open as the toggle was getting confused.

File: engine/app/components/citizens_advice_components/on_this_page.rb (hunk 1 above)
Comment (davidsauntson) [reply engine/app/components/citizens_advice_components/on_this_page.rb:40]: The logic looks OK to me.

You can simplify it a bit by not passing the link into its own `children?` method, and use the instance variable `@children` to see if they are present.

So you would have
```rb 
def children_exist?
  links.any? { |link| link.children? }
end
```
and in `class Link` have
```rb
def children?
   @children.present?
end
```

File: engine/spec/components/citizens_advice_components/on_this_page_spec.rb [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          "Link 5"
+        ]
+      end
+
+      it "renders nested links" do
+        expect(component.at("[data-testid='on-this-page-children']")).not_to be_nil
+      end
+
+      it "renders toggle buttons" do
+        expect(component.at("[data-testid='on-this-page-toggle']")).not_to be_nil
```
Comment (davidsauntson) [inline engine/spec/components/citizens_advice_components/on_this_page_spec.rb:102]: You could make this a bit more specific and assert that the number of toggles present is correct maybe?

**Review by davidsauntson — APPROVED** [review]
Looking great!

Small question about one of the rspecs and a possible simplification of `children_exist` but nothing major 👍🏻

**Comment by davidsauntson** [comment]
OH please can you update the changelog too, thanks 🙏🏻

**Pull request merged**